app/models/problem.rb

```diff
@@ -10,7 +10,7 @@ class Problem < ActiveRecord::Base
   belongs_to :assessment, touch: true
   has_many :annotations
 
-  validates :name, presence: true
+  validates :name, presence: true, uniqueness: {scope: :assessment_id}
```

Comment:
Can you check these files with rubocop?  I think validates with a :uniqueness argument is the preferred way to add this check.  If so, I think the other files should be edited to use that method instead, just to keep the code slightly more up to date with conventions.